fix(compact sensor): correctly parse alphabetic id string modifier

richardstephens opened the pull request:
This should be a simple one, but I have two questions:
1. What do you think about refactoring these to use binary literals instead of hex literals for pulling out the right bits?
2. I don't love the number of cases under which these parse functions panic - Should they return a Result<> instead?

datdenkikniet commented:
1. Binary literals are definitely better for understandability, so I think that's a good idea. 
2. Yeah, I originally went with `Option<T>` because it was just easier to get going. Returning actually descriptive errors is absolutely on the roadmap, but has not been priority for now.

LMK if you would like this PR merged as-is, or if you want to add more changes before doing so.

richardstephens commented:
I've added a changelog entry to this PR and rebased on master. 

I'm looking at implementing some of the other SDR types but that could take a bit longer so I think it makes sense to merge this as-is.

BTW, I sent you an email about test hardware :)
